packages/code-editor/src/Markdown/utils.spec.ts

```diff
@@ -0,0 +1,133 @@
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import {
+  qualifyMarkdownText,
+  getCodeLanguageFromTaggedText,
+  removeCodeLanguageTags,
+  addCodeLanguageTags,
+} from './utils'
+
+export const toMdString = (textLines: string[]) => textLines.join('\n')
+
+describe('Markdown utils', () => {
+  const example = [
+    '# Run the query that is specified inline in the posted body.',
+    'This allows running a query as defined in json in the posted body. This combines the two actions of posting & running a query into one step.',
+    'Here is an example body in json:',
+    '```',
+    '{',
+    '  "model":"thelook",',
+    '  "view":"inventory_items",',
+    '  "filters":{"category.name":"socks"},',
+    '  "sorts":["products.count desc 0"],',
+    '  "limit":"500",',
+    '  "query_timezone":"America/Los_Angeles"',
+    '}',
+    '```',
+    '',
+  ]
+  describe('search highlighting', () => {
+    const str = toMdString(example.slice(0, 2))
+    test('it returns original string when there are no matches', () => {
+      const result = qualifyMarkdownText(str, 'dashboard')
+      expect(result).toEqual(str)
+    })
+
+    test('it returns original string when no pattern is provided', () => {
+      const result = qualifyMarkdownText(str, '')
+      expect(result).toEqual(str)
+    })
+
+    test('it wraps matches with mark tags', () => {
+      const result = qualifyMarkdownText(str, 'query')
+      expect(result).toEqual(
+        toMdString([
+          '# Run the <mark>query</mark> that is specified inline in the posted body.',
+          'This allows running a <mark>query</mark> as defined in json in the posted body. This combines the two actions of posting & running a <mark>query</mark> into one step.',
+        ])
+      )
+    })
+  })
+  describe('syntax highlighting', () => {
+    test('it returns original string when there are no code blocks', () => {
+      const str = toMdString(example.slice(0, 2))
+      const result = qualifyMarkdownText(str, '')
+      expect(result).toEqual(str)
+    })
+    test('it returns original string when code block syntax is unspecified', () => {
```

Comment:
I find it's so much easier for me to read tests when there's some white space between one test and another, and you did it elsewhere hence I thought I'd mention it. Feel free to ignore these
```suggestion

    test('it returns original string when code block syntax is unspecified', () => {
```